chain/gen/gen.go

```diff
@@ -508,68 +530,38 @@ type ProofInput struct {
 	vrfout  []byte
 }
 
-func IsRoundWinner(ctx context.Context, ts *types.TipSet, round int64, miner address.Address, epp ElectionPoStProver, a MiningCheckAPI) (*ProofInput, error) {
+func IsRoundWinner(ctx context.Context, ts *types.TipSet, round abi.ChainEpoch,
+	miner address.Address, brand types.BeaconEntry, a MiningCheckAPI) (*types.ElectionProof, error) {
+
 	buf := new(bytes.Buffer)
 	if err := miner.MarshalCBOR(buf); err != nil {
 		return nil, xerrors.Errorf("failed to cbor marshal address: %w")
 	}
-	epostRand, err := a.ChainGetRandomness(ctx, ts.Key(), crypto.DomainSeparationTag_ElectionPoStChallengeSeed, abi.ChainEpoch(round-build.EcRandomnessLookback), buf.Bytes())
+
+	electionRand, err := store.DrawRandomness(brand.Data, 17, round, buf.Bytes())
 	if err != nil {
-		return nil, xerrors.Errorf("chain get randomness: %w", err)
+		return nil, xerrors.Errorf("failed to draw randomness: %w", err)
 	}
 
 	mbi, err := a.MinerGetBaseInfo(ctx, miner, ts.Key())
 	if err != nil {
 		return nil, xerrors.Errorf("failed to get mining base info: %w", err)
 	}
 
-	vrfout, err := ComputeVRF(ctx, a.WalletSign, mbi.Worker, epostRand)
+	vrfout, err := ComputeVRF(ctx, a.WalletSign, mbi.Worker, electionRand)
 	if err != nil {
 		return nil, xerrors.Errorf("failed to compute VRF: %w", err)
 	}
 
-	if len(mbi.Sectors) == 0 {
-		return nil, nil
-	}
-
-	var sinfos []abi.SectorInfo
-	for _, s := range mbi.Sectors {
-		if s.Info.Info.RegisteredProof == 0 {
-			return nil, xerrors.Errorf("sector %d in proving set had registered type of zero", s.ID)
-		}
-		sinfos = append(sinfos, abi.SectorInfo{
-			SectorNumber:    s.ID,
-			SealedCID:       s.Info.Info.SealedCID,
-			RegisteredProof: s.Info.Info.RegisteredProof,
-		})
-	}
-
-	hvrf := blake2b.Sum256(vrfout)
-	candidates, err := epp.GenerateCandidates(ctx, sinfos, hvrf[:])
-	if err != nil {
-		return nil, xerrors.Errorf("failed to generate electionPoSt candidates: %w", err)
-	}
-
-	var winners []storage.PoStCandidateWithTicket
-	for _, c := range candidates {
-		if types.IsTicketWinner(c.Candidate.PartialTicket, mbi.SectorSize, uint64(len(sinfos)), mbi.NetworkPower) {
-			winners = append(winners, c)
-		}
-	}
-
-	// no winners, sad
-	if len(winners) == 0 {
+	// TODO: wire in real power
```

Comment:
This is the 'quality adjusted power' right?

Reply:
todo was from before I even fetched power but also that ^^